cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java

```diff
@@ -86,20 +83,18 @@ public void deleteCloudWatchAlarmsForSystemFailures(CloudStack stack, String reg
     }
 
     public void deleteCloudWatchAlarmsForSystemFailures(CloudStack stack, String regionName, AwsCredentialView credentialView, List<String> instanceIds) {
-        if (isCloudwatchEnabled(stack)) {
-            List<String> instanceIdsFromStack = stack.getGroups().stream()
-                    .flatMap(group -> group.getInstances().stream())
-                    .map(CloudInstance::getInstanceId)
-                    .collect(Collectors.toList());
-            List<String> instanceIdsNotInStack = instanceIds.stream()
-                    .filter(instanceId -> !instanceIdsFromStack.contains(instanceId))
-                    .collect(Collectors.toList());
-            if (!instanceIdsNotInStack.isEmpty()) {
-                LOGGER.warn("Instance IDs [{}] are not part of cloud stack {}, these instances may have already been deleted on the cloud provider side.",
-                        instanceIdsFromStack, stack);
-            }
-            deleteCloudWatchAlarmsForSystemFailures(regionName, credentialView, instanceIds);
+        List<String> instanceIdsFromStack = stack.getGroups().stream()
```

Comment:
I think instancemetadata also contains this information

Reply:
yes.

Reply:
Okay sorry I am just now really looking into this, but I don't see an easy way to grab the instance metadata from the inputs we have as that doesn't seem to be exposed through the CloudStack object.
Also what benefit do you see from getting that information through there instead?

It appears to me that the original purpose for using the instances directly from the CloudStack was to be able to see which instances were still lively vs already deleted, and then only delete the alarms from the lively ones. Does using the metadata also enable this, and if so does it make this process better?

Reply:
yeah I just checked this is not a trivial task